### PR TITLE
Glob 51422 add diff option

### PR DESCRIPTION
**Pull request by DCGlobality**
Closes [GLOB-51422]

Small/Medium PR that creates --diff argument in globality-black
It uses git diff to report the differences that globality-black would induces if it was applied to the file/directory

[GLOB-51422]: https://globality.atlassian.net/browse/GLOB-51422

### CODE REVIEW

File: globality_black/cli.py [hunk 1]
```diff
@@ -133,8 +145,12 @@ def process_path(
 
     if not check_only_mode:
         path.write_text(output_code)
-
-    return is_modified, False, f"{initial_str} {path}"
+    if diff_mode:
+        # if diff we add the diff report to the reformat message
+        output = diff_output + "\n" + f"{initial_str} {path}"
+    else:
+        output = f"{initial_str} {path}"
```
Comment (marcglobality) [inline globality_black/cli.py:152]:
```suggestion
    output_msg = f"{initial_str} {path}"
    if diff_mode:
        # if diff we add the diff report to the beginning of the reformat message
        output_msg = diff_output + "\n" + output_msg
```

File: globality_black/diff.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    Use git diff to report the differences that globality black
+    will make to input_path file
+    pexpect is used over subprocess because of better color handling
+    """
+
+    with tempfile.NamedTemporaryFile(mode="w+t") as temp_file:
+        temp_file.writelines(output_code)
+        temp_file.seek(0)
+        output = pexpect.run(
+            "git --no-pager diff --no-index {} {}".format(input_path, temp_file.name)
```
Comment (marcglobality) [inline globality_black/diff.py:18]: why not f-string?

File: globality_black/tests/test_cli.py [hunk 3]
```diff
@@ -28,10 +28,12 @@ class FileCondition(Enum):
 }
 
 
-@pytest.mark.parametrize("check", (False, True))
+@pytest.mark.parametrize("check,diff", [(False, False), (True, False), (True, True)])
```
Comment (marcglobality) [inline globality_black/tests/test_cli.py:31]: maybe add a comment here to remind why False, True is not possible. We know now but will forget soon :)

File: globality_black/tests/test_cli.py [hunk 4]
```diff
@@ -28,10 +28,12 @@ class FileCondition(Enum):
...
 @pytest.mark.parametrize("verbose", (False, True))
 @pytest.mark.parametrize("file_condition", tuple(FileCondition))
-def test_cli_file(runner: CliRunner, check: bool, verbose: bool, file_condition: FileCondition):
+def test_cli_file(
+    runner: CliRunner, check: bool, verbose: bool, diff: bool, file_condition: FileCondition
```
Comment (marcglobality) [inline globality_black/tests/test_cli.py:35]: add magic comma so the params are exploded

File: globality_black/tests/test_cli.py [hunk 5]
```diff
@@ -152,7 +168,7 @@ def test_cli_directory(runner: CliRunner, check: bool, error: bool, verbose: boo
             shutil.copy(str(fixture_input_path), str(filename_in_temp))
             filenames_in_temp.append(filename_in_temp)
 
-        result = run_globality_black(runner, temp_path, check, verbose)
+        result = run_globality_black(runner, temp_path, check, verbose, diff=False)
```
Comment (marcglobality) [inline globality_black/tests/test_cli.py:171]: the default is False, why you need to pass this?

File: globality_black/cli.py (hunk 1 above)
Comment (marcglobality) [reply globality_black/cli.py:152]: I would also add a "\n" or maybe "------------" before `diff_output`. It's a bit difficult to separte the blocks for each files visually and find the filenames when you run in a repo . WDYT?

**Review by marcglobality — APPROVED** [review]
I love this, thanks! Added some nit comments and also a functional one I think it's important so repeating here

I would also add a "\n" or maybe "------------" before `diff_output`. It's a bit difficult to separte the blocks for each files visually and find the filenames when you run in a repo . WDYT?

**Review by tomaszpietruszka-globality — APPROVED** [review]
I second Marc's comments, otherwise LGTM

**Pull request merged**